src/test/unit_tests/test_nsfs_access.js

```diff
@@ -99,6 +109,24 @@ mocha.describe('new tests check', async function() {
             assert.equal(err.code, 'EACCES');
         }
     });
+
+    mocha.it('NON ROOT 3 with suplemental group - success', async function() {
+        await nb_native().fs.mkdir(NON_ROOT1_FS_CONFIG, full_path_non_root1, 0o770);
+        //non root3 has non-root1 group as supplemental group, so it should succeed
+        const non_root_entries = await nb_native().fs.readdir(NON_ROOT3_FS_CONFIG, full_path_non_root1);
+        assert.equal(non_root_entries && non_root_entries.length, 0);
+    });
+
+    mocha.it('NON ROOT 3 without the files gid inside the suplemental group - failure', async function() {
+        await nb_native().fs.mkdir(NON_ROOT2_FS_CONFIG, full_path_non_root2, 0o770);
+        try {
+            //non root3 doesn't have non-root1 group as supplemental group, so it should fail
```

Comment:
I think there is a typo in the comment
```suggestion
            //non root3 doesn't have non-root2 group as supplemental group, so it should fail
```